cmd/erasure-metadata.go

```diff
@@ -329,12 +335,12 @@ func objectQuorumFromMeta(ctx context.Context, partsMetaData []FileInfo, errs []
 		return 0, 0, err
 	}
 
-	dataBlocks := latestFileInfo.Erasure.DataBlocks
 	parityBlocks := globalStorageClass.GetParityForSC(latestFileInfo.Metadata[xhttp.AmzStorageClass])
 	if parityBlocks <= 0 {
 		parityBlocks = defaultParityCount
 	}
 
+	dataBlocks := len(partsMetaData) - parityBlocks
```

Comment:
@vadmeste found the real reason for `dataBlocks` to be `0` - it should never be set to `0` this is a bug that has been present since versioning was introduced :-).

Reply:
> @vadmeste found the real reason for `dataBlocks` to be `0` - it should never be set to `0` this is a bug that has been present since versioning was introduced :-).

Well it is for delete marker, since delete marker doesn't have data, erasure is not used & data + parity won't make sense here.